pydantic/utils.py

```diff
@@ -268,3 +269,16 @@ def resolve_annotations(raw_annotations: Dict[str, AnyType], module_name: Option
 
 def is_callable_type(type_: AnyType) -> bool:
     return type_ is Callable or getattr(type_, '__origin__', None) is Callable
+
+
+def _check_classvar(v: AnyType) -> bool:
+    return type(v) == type(ClassVar) and (sys.version_info < (3, 7) or getattr(v, '_name', None) == 'ClassVar')
+
+
+def is_classvar(ann_type: AnyType) -> bool:
+    return _check_classvar(ann_type) or _check_classvar(getattr(ann_type, '__origin__', None))
+
+
+def gather_validators(type_: AnyType) -> Dict[str, classmethod]:
```

Comment:
this should be in `class_validators`, also can we use the same logic for models and dataclasses, without making it too complicated?

Reply:
The logic for models (extract_validators function ) is still used by dataclasses when the __pydantic__model is created. The new gather_validators function just gets the validators in a format to where they can be added to the namespace in the create_model function. Then the same logic as for BaseModel is used when the Model is created.

Both extract_validators and gather_validators contain a check for the "__validator_config" attribute so there is a slight duplication there.  One alternative would be to add the entire namespace (__dict__) of the dataclass to create_model (instead of just validators) and then let the __new__ method filter the namespace for validators as normal. What do you think?